docs/reference/extensions/flask-framework.rst

```diff
@@ -49,6 +63,20 @@ Some files, if they exist, are included by default. These include:
 ``app``, ``app.py``, ``migrate``, ``migrate.sh``, ``migrate.py``, ``static``,
 ``templates``.
 
+**Regarding the migrate.sh file:**
+
+If your app depends on a database it is common to run a database migration
+script before app startup which, for example, creates or modifies tables.
+This can be done by including the ``migrate.sh`` script in the root of your
+project. It will be executed with the same environment variables and context
+as the Flask application.
+
+If the migration script fails, the app won't be started and the app charm
+will go into blocked state. The migration script will be run on every unit
+and it is assumed that it is idempotent (can be run multiple times) and that
+it can be run on multiple units at the same time without causing issues.
+This can be achieved by, for example, locking any tables during the migration.
+
```

Comment:
This is only true when the Rock is running within the Charm as the charm looks for this file and executes it. The Rock does not execute the migration on its own

Reply:
Thanks, that's good to know. So should this note be moved into the reference page for the Charmcraft extension?

Reply:
yes please